dvc/remote/s3.py

```diff
@@ -207,10 +207,24 @@ def _list_paths(self, path_info, max_items=None):
     def list_cache_paths(self):
         return self._list_paths(self.path_info)
 
+    def isfile(self, path_info):
+        if path_info.path.endswith("/"):
+            return False
+
+        try:
+            self.get_head_object(self.s3, path_info.bucket, path_info.path)
+        except DvcException:
```

Comment:
We should catch a more specific exception here, meaning use `s3.head_object()` directly.

Reply:
Actually that one was fine too as it is the only DvcException that it raises :) But the current one is also fine.